### PR TITLE
Stop tracking native view controllers on iOS

### PR DESCRIPTION
**Pull request by bnickel**
This change disables view controller capture on iOS unless `enableNativeTouchEventCapture` is set.  This prevents a flood of meaningless `RNSScreen` pageviews that cause confusion during analysis.

I also updated `Heap.setAppId` to use the same initialization logic as auto-init.

To do this, Heap sets the new `HeapOptions.disableViewControllerAutocapture` option to `YES` on initialize.  Each time Heap needs a new pageview (once per session) an empty one is added to the queue.

## Test Plan

Changes were tested manually in a demo app and validated with detox.

## Checklist
- [x] Detox tests pass
- [x] If this is a bugfix/feature, the changelog has been updated

### CODE REVIEW

File: ios/RNHeapInit.m [hunk 1]
```diff
@@ -17,27 +17,39 @@ + (void)load {
...
-    NSLog(@"Auto-initializing the Heap library with app ID %@ with native autocapture enabled=%@.", appId, @(enableTouchAutocapture));
-    
+    NSLog(@"Auto-initializing the Heap library with app ID %@ with native autocapture enabled=%@.", appId, enableTouchAutocapture ? @"true": @"false");
+
+    [self initializeWithAppId:appId enableTouchAutocapture:enableTouchAutocapture enableDebugLogging:enableDebugLogging captureBaseUrl:captureBaseUrl];
+}
+
++ (void)initializeWithAppId:(NSString *)appId enableTouchAutocapture:(BOOL)enableTouchAutocapture enableDebugLogging:(BOOL)enableDebugLogging captureBaseUrl:(NSURL *)captureBaseUrl
+{
+    static BOOL alreadyInitialized = NO;
```
Comment (apyousef) [inline ios/RNHeapInit.m:37]: How does this get set to `YES`?

**Review by apyousef — COMMENTED** [review]
@bnickel I had one question.

**Comment by bnickel** [comment]
@apyousef This is ready for re-review.

**Review by apyousef — APPROVED** [review]
@bnickel LGTM

**Pull request merged**